tests/admin/test_dao.py

```diff
@@ -86,7 +177,39 @@ def test_dao_revoke_admin_role_to_valid_user(self):
         user = UserModel.query.filter_by(id=2).first()
         self.assertFalse(user.is_admin)
 
+    
+    """
+    Checks whether a user is trying to revoke other user's admin priviledges. 
+    """
+    def test_dao_revoke_admin_role_by_non_admin_user(self):
+        
+        dao = AdminDAO()
+
+        user = UserModel(
+            name=user1['name'],
+            username=user1['username'],
+            email=user1['email'],
+            password=user1['password'],
+            terms_and_conditions_checked=user1['terms_and_conditions_checked']
+        )
+        user.is_email_verified = True
+        user.save_to_db()
+        user = UserModel.query.filter_by(id=2).first()
+        self.assertFalse(user.is_admin)
+        
+        data = dict(
+            user_id=1
+        )
+        dao_result = dao.revoke_admin_user(2, data)
+
+        self.assertEqual((messages.USER_ASSIGN_NOT_ADMIN, 403), dao_result)
```

Comment:
```suggestion
        self.assertEqual((messages.USER_REVOKE_NOT_ADMIN, 403), dao_result)
```
Once you fix the message being returned for revoking an admin, you'll have to fix this here